DQMOffline/Trigger/plugins/NoBPTXMonitor.cc

```diff
@@ -0,0 +1,256 @@
+#include "DQMOffline/Trigger/plugins/NoBPTXMonitor.h"
+
+// -----------------------------
+//  constructors and destructor
+// -----------------------------
+
+NoBPTXMonitor::NoBPTXMonitor( const edm::ParameterSet& iConfig ) : 
+  folderName_             ( iConfig.getParameter<std::string>("FolderName") )
+  , jetToken_             ( mayConsume<reco::CaloJetCollection>      (iConfig.getParameter<edm::InputTag>("jets")      ) )   
+  , muonToken_             ( mayConsume<reco::TrackCollection>       (iConfig.getParameter<edm::InputTag>("muons")     ) )   
+  , jetE_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("jetEBinning") )
+  , jetE_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("jetEPSet")    ) )
+  , muonPt_variable_binning_ ( iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<std::vector<double> >("muonPtBinning") )
+  , muonPt_binning_          ( getHistoPSet   (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("muonPtPSet")    ) )
+  , ls_binning_           ( getHistoLSPSet (iConfig.getParameter<edm::ParameterSet>("histoPSet").getParameter<edm::ParameterSet>   ("lsPSet")     ) )
+  , num_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("numGenericTriggerEventPSet"),consumesCollector(), *this))
+  , den_genTriggerEventFlag_(new GenericTriggerEventFlag(iConfig.getParameter<edm::ParameterSet>("denGenericTriggerEventPSet"),consumesCollector(), *this))
+  , jetSelection_ ( iConfig.getParameter<std::string>("jetSelection") )
+  , muonSelection_ ( iConfig.getParameter<std::string>("muonSelection") )
+  , njets_      ( iConfig.getParameter<int>("njets" )      )
+  , nmuons_     ( iConfig.getParameter<int>("nmuons" )     )
+{
+
+  jetENoBPTX_.numerator   = nullptr;
+  jetENoBPTX_.denominator = nullptr;
+  jetENoBPTX_variableBinning_.numerator   = nullptr;
+  jetENoBPTX_variableBinning_.denominator = nullptr;
+  jetEVsLS_.numerator   = nullptr;
+  jetEVsLS_.denominator = nullptr;
+
```

Comment:
in addition, shouldn't the jet depend on HEP17 ?
I think you should add a 2D map, at least